cmd/deploy/command_configuration.go

```diff
@@ -178,9 +180,8 @@ func switchRepoSchemaToHTTPS(repo string) (*url.URL, error) {
 		return repoURL, nil
 	case httpsScheme:
 		return repoURL, nil
-	default:
-		return nil, errUnsupportedScheme
 	}
+	return nil, nil
```

Comment:
I think the goal is to don't return any error. We can return just nil in line 174 and make that the function `switchRepoSchemaToHTTPS` returns just a pointer to a URL. If there is an error parsing the URL, we just return nil.

In that way, we can simplify a bit the code because we can change the function `addEnvVars` to not return an error. This case was the only one generating that scenario

Reply:
It would be something like this:

```
diff --git a/cmd/deploy/command_configuration.go b/cmd/deploy/command_configuration.go
index 6489ce2b..470e11dc 100644
--- a/cmd/deploy/command_configuration.go
+++ b/cmd/deploy/command_configuration.go
@@ -114,7 +114,7 @@ func mergeServicesToDeployFromOptionsAndManifest(deployOptions *Options) {
        }
 }

-func addEnvVars(ctx context.Context, cwd string) error {
+func addEnvVars(ctx context.Context, cwd string) {
        if os.Getenv(model.OktetoGitBranchEnvVar) == "" {
                branch, err := utils.GetBranch(cwd)
                if err != nil {
@@ -130,11 +130,12 @@ func addEnvVars(ctx context.Context, cwd string) error {
                }

                if repo != "" {
-                       repoHTTPS, err := switchRepoSchemaToHTTPS(repo)
-                       if err != nil {
-                               return err
+                       repoHTTPS := switchRepoSchemaToHTTPS(repo)
+                       if repoHTTPS == nil {
+                               repo = ""
+                       } else {
+                               repo = repoHTTPS.String()
                        }
-                       repo = repoHTTPS.String()
                }
                os.Setenv(model.GithubRepositoryEnvVar, repo)
        }
@@ -163,23 +164,22 @@ func addEnvVars(ctx context.Context, cwd string) error {
                os.Setenv(model.OktetoTokenEnvVar, okteto.Context().Token)
        }
        oktetoLog.AddMaskedWord(os.Getenv(model.OktetoTokenEnvVar))
-       return nil
 }

-func switchRepoSchemaToHTTPS(repo string) (*url.URL, error) {
+func switchRepoSchemaToHTTPS(repo string) *url.URL {
        repoURL, err := giturls.Parse(repo)
        if err != nil {
-               return nil, err
+               return nil
        }
        switch repoURL.Scheme {
        case sshScheme, httpScheme:
                repoURL.Scheme = httpsScheme
                repoURL.User = nil
-               return repoURL, nil
+               return repoURL
        case httpsScheme:
-               return repoURL, nil
+               return repoURL
        default:
-               return nil, errUnsupportedScheme
+               return nil
        }
 }

diff --git a/cmd/deploy/deploy.go b/cmd/deploy/deploy.go
index 7c457566..36c4c313 100644
--- a/cmd/deploy/deploy.go
+++ b/cmd/deploy/deploy.go
@@ -266,9 +266,7 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
                return err
        }

-       if err := addEnvVars(ctx, cwd); err != nil {
-               return err
-       }
+       addEnvVars(ctx, cwd)
        oktetoLog.Debugf("creating temporal kubeconfig file '%s'", dc.TempKubeconfigFile)
        if err := dc.Kubeconfig.Modify(dc.Proxy.GetPort(), dc.Proxy.GetToken(), dc.TempKubeconfigFile); err != nil {
                oktetoLog.Infof("could not create temporal kubeconfig %s", err)
```